runtime/compilers/rillv1/parser.go

```diff
@@ -414,7 +428,7 @@ func (p *Parser) reparseExceptRillYAML(ctx context.Context, paths []string) (*Di
 			oldDotEnv := p.DotEnv
 			err := p.parseDotEnv(ctx, checkPaths[i])
 			if err == nil {
-				modifiedDotEnv = !maps.Equal(p.DotEnv, oldDotEnv)
+				modifiedDotEnv = !reflect.DeepEqual(p.DotEnv, oldDotEnv)
```

Comment:
I don't think this works:
1. Previously `parseDotEnv` overwrote `p.DotEnv`, so `oldDotEnv` would be a different pointer. But now `parseDotEnv` adds to `p.DotEnv`, so with `oldDotEnv` being a pointer, the equality check will always return `true`.
2. Now that there can be multiple `.env` files, I don't think it's safe to overwrite `modifiedDotEnv` each time a `.env` file is parsed.